common/asyncworkflow/queue/consumer/default_consumer.go

```diff
@@ -0,0 +1,262 @@
+			logger.Error("Failed to nack message", tag.Error(err))
+		}
+		return
+	}
+
+	if err := msg.Ack(); err != nil {
+		logger.Error("Failed to ack message", tag.Error(err))
+	}
+	logger.Debug("Processed message successfully")
+}
+
+func (c *DefaultConsumer) processRequest(logger log.Logger, request *sqlblobs.AsyncRequestMessage) error {
+	switch request.GetType() {
+	case sqlblobs.AsyncRequestTypeStartWorkflowExecutionAsyncRequest:
+		startWFReq, err := decodeStartWorkflowRequest(request.GetPayload(), request.GetEncoding())
+		if err != nil {
+			c.scope.IncCounter(metrics.AsyncWorkflowFailureCorruptMsgCount)
+			return err
+		}
+
+		scope := c.scope.Tagged(metrics.DomainTag(startWFReq.GetDomain()))
+
+		var resp *types.StartWorkflowExecutionResponse
+		op := func() error {
+			resp, err = c.frontendClient.StartWorkflowExecution(c.ctx, startWFReq)
+			return err
+		}
+
+		if err := callFrontendWithRetries(c.ctx, op); err != nil {
+			scope.IncCounter(metrics.AsyncWorkflowFailureByFrontendCount)
+			return fmt.Errorf("start workflow execution failed after all attempts: %w", err)
+		}
+
+		scope.IncCounter(metrics.AsyncWorkflowSuccessCount)
+		logger.Info("StartWorkflowExecution succeeded", tag.WorkflowID(startWFReq.GetWorkflowID()), tag.WorkflowRunID(resp.GetRunID()))
+	case sqlblobs.AsyncRequestTypeSignalWithStartWorkflowExecutionAsyncRequest:
+		startWFReq, err := decodeSignalWithStartWorkflowRequest(request.GetPayload(), request.GetEncoding())
+		if err != nil {
+			c.scope.IncCounter(metrics.AsyncWorkflowFailureCorruptMsgCount)
+			return err
+		}
+
+		scope := c.scope.Tagged(metrics.DomainTag(startWFReq.GetDomain()))
+
+		var resp *types.StartWorkflowExecutionResponse
+		op := func() error {
+			resp, err = c.frontendClient.SignalWithStartWorkflowExecution(c.ctx, startWFReq)
+			return err
+		}
+
+		if err := callFrontendWithRetries(c.ctx, op); err != nil {
+			scope.IncCounter(metrics.AsyncWorkflowFailureByFrontendCount)
+			return fmt.Errorf("signal with start workflow execution failed after all attempts: %w", err)
+		}
+
+		scope.IncCounter(metrics.AsyncWorkflowSuccessCount)
+		logger.Info("SignalWithStartWorkflowExecution succeeded", tag.WorkflowID(startWFReq.GetWorkflowID()), tag.WorkflowRunID(resp.GetRunID()))
+	default:
+		c.scope.IncCounter(metrics.AsyncWorkflowSuccessCount)
+		return &UnsupportedRequestType{Type: request.GetType()}
+	}
+
+	return nil
+}
+
+func callFrontendWithRetries(ctx context.Context, op func() error) error {
+	throttleRetry := backoff.NewThrottleRetry(
+		backoff.WithRetryPolicy(common.CreateFrontendServiceRetryPolicy()),
+		backoff.WithRetryableError(common.IsServiceTransientError),
+	)
+
+	return throttleRetry.Do(ctx, op)
+}
+
+func decodeStartWorkflowRequest(payload []byte, encoding string) (*types.StartWorkflowExecutionRequest, error) {
+	if encoding != string(common.EncodingTypeJSON) {
+		return nil, &UnsupportedEncoding{EncodingType: encoding}
+	}
+
+	var startRequest types.StartWorkflowExecutionRequest
```

Comment:
The payload is actually the async request. Although it only has one field which is the sync request today, I'm not sure if we'll add other fields to async requests such as retry policy.